core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -194,18 +215,73 @@ public void accept(Object o) throws Throwable {
 
             if (logFile != null) {
                logFile.close();
-               completeFile(logFile.fileId, currentOffset);
                logFile = null;
+               nextExpirationTime = -1;
+            }
+         }
+         return;
+      }
+
+      if (o instanceof CompactionExpirationSubscriber) {
```

Comment:
Wouldn't it be much cleaner to introduce a Class or enum for the different events that are being accepted by the compactor instead of `Object`?

Reply:
Unfortunately,  I cannot use an enum as I need to have a different value for each invocation. We could add a class to do it, but I figured it wasn't really needed. Do you think we should add it? There are 3 cases, RESUME which already has an explicit object, the expiration subscriber and an int. They all seemed pretty distinct to me, but if you feel it makes it more complicated we can change.

Reply:
I guess it's fine as it is as the logic to distinguish this sate isn't required elsewhere, only in `accept`.

Reply:
It is, and the only one who can pass values directly to it is the Compactor as well, so everything is isolated to one file and the different types are easily seen in the accept method.